plugins/modules/rds_instance_snapshot.py

```diff
@@ -376,20 +338,15 @@ def main():
         argument_spec=argument_spec,
         supports_check_mode=True,
     )
-
-    retry_decorator = AWSRetry.jittered_backoff(retries=10)
-    try:
-        client = module.client("rds", retry_decorator=retry_decorator)
-    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
-        module.fail_json_aws(e, msg="Failed to connect to AWS.")
+    client = module.client("rds", retry_decorator=AWSRetry.jittered_backoff())
```

Comment:
Is there any reason to remove `try - except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:` ?

Reply:
Strictly speaking module.client can still pass through a couple of BotoCoreError exceptions (I don't think it can currently pass a ClientError):

- `UnknownServiceError`
- `InvalidDefaultsMode`
- `ConfigNotFound`

I think it might make more sense to tweak `boto3_conn` (plugin_utils and module_utils) to catch the broader BotoCoreError and drop the logic here.

Reply:
That makes more sense.

Reply:
Will update!

Reply:
#2157 Would remove the need for the try/catch at this point.

Reply:
@tremble so should I undo my latest commit here or just leave it so we can merge this pr?